src/summary.cpp

```diff
@@ -0,0 +1,140 @@
+#include <lo2s/config.hpp>
+#include <lo2s/summary.hpp>
+#include <lo2s/util.hpp>
+
+#include <boost/filesystem.hpp>
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <ratio>
+#include <array>
+
+extern "C" {
+#include <sys/time.h>
+}
+
+namespace lo2s
+{
+
+Summary& summary()
+{
+    static Summary s;
+    return s;
+}
+
+Summary::Summary()
+: start_wall_time_(std::chrono::steady_clock::now()), num_wakeups_(0),
+  thread_count_(0), trace_dir_(""), exit_code_(0)
+{
+}
+
+void Summary::register_process(pid_t pid)
+{
+    pids_.emplace(pid);
+}
+
+void Summary::record_perf_wakeups(std::size_t num_wakeups)
+{
+    num_wakeups_ += num_wakeups;
+}
+
+void Summary::set_exit_code(int exit_code)
+{
+    exit_code_ = exit_code;
+}
+
+void Summary::add_thread()
+{
+    thread_count_++;
+}
+
+std::string pretty_print_bytes(std::size_t trace_size)
+{
+    double result_size = trace_size;
+    int unit = 0;
+
+    std::ostringstream out;
+    std::array<std::string, 5> units = { "B", "KiB", "MiB", "GiB", "TiB" };
+
+    while(result_size > 1024)
+    {
+        result_size /= 1024;
+        unit++;
+
+        //We can not get higher than TiB so break here
+        if(unit == units.size() - 1)
+        {
+            break;
+        }
+    }
+
+    out << std::fixed << std::setprecision(2) << result_size << units[unit];
+    return out.str();
+}
+
+void Summary::set_trace_dir(const std::string& trace_dir)
+{
+    trace_dir_ = trace_dir;
+}
+
+void Summary::show()
+{
+    std::chrono::duration<double> wall_time, cpu_time;
```

Comment:
Same as above, please place variable declarations as close as possible to the first use. Also, never ever declare two variables on one line.